### ISSUE TITLE
Focus trap doesn't work

### ISSUE THREAD

**Issue by alex996** [discord/focus-layers#18]
Thanks for your work on this library. Unfortunately, when I tried running the demo from the readme, I couldn't get it to work. The focus does get trapped on `Shift + Tab` but when `Tab` is pressed, I'm able to escape the lock and circle around the page.

Here's a [CodeSandbox](https://codesandbox.io/s/relaxed-mclean-hcvxf?file=/src/App.js) with the copy-paste code, and here it is in a [separate tab](https://hcvxf.csb.app/) for testing. To repro:
1. Tab into the "Show Dialog" button
2. Press `Enter` to open the hidden div
2. Press `Tab` repeatedly

It's expected that the focus stays on the "Close Dialog" button. In reality, the focus moves around the page (to search bar, to "Show Dialog", etc.).

**Comment by faultyserver** [comment]
Hi! This is a good catch that should probably get mentioned earlier in the docs.

The reason the focus is not getting trapped properly is because there is no tabbable element within the document that appears _after_ the Dialog in the DOM, and for some reason there isn't a clean way to intercept that kind of blur event (at least that I've been able to find). This is why `shift+tab` is handled appropriately (the open button is there) but `tab` is not.

The appropriate workaround here is to add `FocusGuard`s around your app to make sure there is _always_ a tabbable element both before and after your Dialog. This is explained in more detail in the [Edge Guards section of the README](https://github.com/discord/focus-layers#edge-guards), but this is a good indicator that it should be moved higher into a more general usage example at the top.

With this change, the `App` component in that sandbox could look something like this:

```typescript
import useFocusLock, {FocusGuard} from "focus-layers";

export default function App() {
  const [open, setOpen] = React.useState(false);

  return (
    <div>
      <FocusGuard />
      <button onClick={() => setOpen(true)}>Show Dialog</button>
      {open && (
        <Dialog>
          <p>
            When this dialog is open, focus is trapped inside! Tabbing will
            always bring focus back to this button.
          </p>
          <button onClick={() => setOpen(false)}>Close Dialog</button>
        </Dialog>
      )}
      <FocusGuard />
    </div>
  );
}
```

**Comment by alex996** [comment]
@faultyserver Thanks for the detailed explanation! It works as expected with `FocusGuard`. I think mentioning this workaround under [Basic Usage](https://github.com/discord/focus-layers#basic-usage) is a great idea :+1:
